common/src/main/java/io/pravega/common/util/Retry.java

```diff
@@ -248,10 +250,10 @@ boolean canRetry(final Throwable e) {
         }
 
         private Class<? extends Throwable> getErrorType(final Throwable e) {
-            if (retryType.equals(CompletionException.class) || throwType.equals(CompletionException.class)) {
+            if (ExceptionHelpers.shouldUnwrap(retryType) || ExceptionHelpers.shouldUnwrap(throwType)) {
                 return e.getClass();
             } else {
-                if (e instanceof CompletionException && e.getCause() != null) {
+                if ((e instanceof CompletionException || e instanceof ExecutionException) && e.getCause() != null) {
```

Comment:
maybe create a `shouldUnwrap` that takes in a `Throwable` argument? At least so we keep those checks close to each other.